src/avt/Queries/Queries/avtXRayImageQuery.C

```diff
@@ -1501,7 +1504,7 @@ avtXRayImageQuery::Execute(avtDataTree_p tree)
             baseName << "output";
 
         // does NOT contain the file extension
-        std::string out_filename_w_path{(outputDir == "." ? "" : outputDir + "/") + baseName.str()};
+        std::string out_filename_w_path{(outputDir == "." ? "" : outputDir + VISIT_SLASH_STRING) + baseName.str()};
```

Comment:
That looks pretty bad; I'll investigate that inside conduit.

Reply:
I'm not sure why the path is getting duplicated like that. The error Conduit is throwing is because of the ":" in the file path, which it is interpreting as a subpath. I will discuss with @cyrush.

Reply:
Without VISIT_SLASH_STRING it worked fine? That seems strange to me.

Reply:
@biagas can you give me the output file that was generated from the query in the failing test?

Reply:
There is one in this [comment.](https://github.com/visit-dav/visit/pull/19106#issuecomment-1834603713)

Reply:
Sorry @biagas, I meant a Blueprint output file. It should have a `.root` extension. You may need to run with `--no-cleanup`.

Reply:
I'm looking at _run/_queries_xrayimage/testdir, there are several subdirs, all with an 'output.root', which subdir?

Reply:
Could you send me the contents of the `hdf5` directory?

Reply:
[output.zip](https://github.com/visit-dav/visit/files/13729351/output.zip)

Reply:
Hi @biagas. I don't think the output file is the problem. I'm guessing there is something wrong with the path that is generated for opening the file. I have added some print statements to the test; could you please (when you get a chance) run the test in verbose mode and report back to me what the messages say for "This is conduit_db outside the for loop:" and then much earlier for "This is conduit_db inside the for loop:". Thank you very much, and sorry for going back and forth on this.

Reply:
Here's the output. The error message is new:
```

This is conduit_db outside the for loop:
C:\A_Visit\3.4RC\single_test_output\_run\_queries_xrayimage\testdir\hdf5\output.root

==============================
Traceback (most recent call last):
  File "C:/A_Visit/3.4RC/visit/src/test/tests/queries/xrayimage.py", line 909, in <module>
    blueprint_test("hdf5", conduit_dir_hdf5, 32, "Blueprint_HDF5_X_Ray_Output")
  File "C:/A_Visit/3.4RC/visit/src/test/tests/queries/xrayimage.py", line 891, in blueprint_test
    test_bp_data(testname + str(i), conduit_db, qro) # bounds
  File "C:/A_Visit/3.4RC/visit/src/test/tests/queries/xrayimage.py", line 613, in test_bp_data
    conduit.relay.io.blueprint.load_mesh(xrayout, conduit_db)
OSError:
file: C:\A_Visit\ThirdParty\conduit-v0.8.7\src\libs\relay\conduit_relay_io_handle.cpp
line: 202
message:
IOHandle does not support opening paths with subpaths specified: "C:\A_Visit\3.4RC\single_test_output\_run\_queries_xrayimage\testdir\hdf5/C:\A_Visit\3.4RC\single_test_output\_run\_queries_xrayimage\testdir\hdf5\output.root"
```

At one point I had this working, with a slight modification to one of your changes.

Reply:
Is there something different for the message for "This is conduit_db inside the for loop:"? That should appear much earlier in the test execution.

Reply:
Sorry about that. 

```
This is conduit_db inside the for loop:
C:\A_Visit\3.4RC\single_test_output\_run\_queries_xrayimage\testdir\hdf5\output.root
```

Reply:
It does seem like a conduit issue, not a visit issue. 

The only think I can think of is that could be a case where we are trying to construct the full path and it's mistaking what is already an absolute path for a relative path in load_mesh.

I'll try the example data a similar full path arg on windows and see what I can find.

Reply:
IIRC I have seen errors like this (extra path information prepended to actual file path) when the Reader (or somewhere else in VisIt or a library) thinks that the passed in filename is not already a fully qualified path, and it attempts to add the current working directory to the filename.  This generally means that internally there is something that is not Windows-aware when trying to determine if a path is fully qualified.

@JustinPrivitera If I modify avtXRayImageQuery line 1507 to use "/" instead of VISIT_SLASH_STRING, then the test works correctly.

Reply:
I have found C++17 std::filesystem::path helpful in dealing with some cross-platform path issues on Windows, but we don't yet require C++17 throughout VisIt.

Reply:
That seems so strange to me, that VISIT_SLASH_STRING would be the culprit. That variable is used by all of the output types for their file name. I think they really ought to use the platform-specific slash string for writing. Perhaps we can see if there is a bug in Conduit that is causing this.

Reply:
I can't open the zip file on my windows machine, or in my linux docker images:

```
Archive:  output.zip
   skipping: output.root             need PK compat. v6.3 (can do v4.6)
```

I can open it on my mac, I'll try to repost the file here.

Reply:
Sorry about that, I use 7z for all my compression needs on Windows.

Reply:
[Uploading output.root.txt…]()

Reply:
I tried to replicate the open error in VisIt 3.3.3 python cli, which is also using conduit 0.8.7 and the path works file. 

```
import conduit
dbfile = "C:\\Users\\cyrus\\Downloads\\output\\output.root"
print(dbfile)
n = conduit.Node()
conduit.relay.io.blueprint.load_mesh(n, dbfile)
print(n)
```

Inside the file -- there are not any absolute paths. 

Maybe there is something subtle about how is being created in the test suite that I don't understand. 

If I give it the funky concatenated path it will fail which we expect --  with the proper message that it can't open the root file  -- not that the IOHandle has an issue.

Reply:
I'm thinking somewhere in the test the escaped path-separators are being stripped from the 'filenames' query results. The 'filenames' may have `\\` to start with, but then in some function they get stripped to just `\`, then they are treated as escaping the next character and things like `\t` become a tab.

Take a look here where I had to go to extreme lengths to ensure the end result for a data path had escaped path-separators on Windows.
https://github.com/visit-dav/visit/blob/5471aaa3c987662ce6a22b59f4eb18fdf56961b9/src/test/py_src/visit_test_main.py#L136

Reply:
We can also use raw strings in python:

```
r"C:\"
```
etc.

Usually the escapes are needed string literals, not necessary constructed paths. 

But I don't see where the double concatenation of the prefix can happen.

If we print the path before opening we should be able to see, we should be able t reproduce on windows outside of the test suite.